src/pages/ProductDetails.jsx

```diff
@@ -31,11 +31,13 @@ export default class ProductDetails extends React.Component {
   }
 
   async componentDidMount() {
-    const { productId, category, title } = this.props.match.params;
+    const { productId, category, title } = this.props.match.params; //.location.state
```

Comment:
ficou um comentariozinho aqui

```suggestion
    const { productId, category, title } = this.props.match.params;
```